File: components/credentials/src/main/java/org/cloudfoundry/credhub/views/CertificateCredentialView.java
package org.cloudfoundry.credhub.views;

import java.util.List;
import java.util.UUID;

import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonProperty;

public class CertificateCredentialView {

  private String name;
  private UUID uuid;
  private List<CertificateVersionView> certificateVersionViews;
  private String signedBy;

  public CertificateCredentialView() {
    super(); /* Jackson */
  }

  public CertificateCredentialView(
    final String name,
    final UUID uuid,
    @NotNull final List<CertificateVersionView> certificateVersionViews,
    final String signedBy) {
    super();
    this.name = name;
    this.uuid = uuid;
    this.certificateVersionViews = certificateVersionViews;
    this.signedBy = signedBy;
  }

  @JsonProperty("id")
  public UUID getUUID() {
    return uuid;
  }

  @JsonProperty("name")
  public String getName() {
    return name;
  }

  @JsonProperty("versions")
  public List<CertificateVersionView> getCertificateVersionViews() {
    return certificateVersionViews;
  }

  @JsonProperty("signed_by")
  public String getSignedBy() {
    return signedBy;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/views/CertificateCredentialView.kt
package org.cloudfoundry.credhub.views

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.UUID
import javax.validation.constraints.NotNull

class CertificateCredentialView {
    @get:JsonProperty("name")
    var name: String? = null
        private set

    @get:JsonProperty("id")
    var uUID: UUID? = null
        private set

    @get:JsonProperty("versions")
    var certificateVersionViews: List<CertificateVersionView>? = null
        private set

    @get:JsonProperty("signed_by")
    var signedBy: String? = null
        private set

    @get:JsonProperty("signs")
    var signs: List<String>? = null
        private set

    constructor() : super() {}
    constructor(
        name: String?,
        uuid: UUID?,
        certificateVersionViews: @NotNull MutableList<CertificateVersionView>?,
        signedBy: String?,
        signs: List<String>?,
    ) : super() {
        this.name = name
        uUID = uuid
        this.certificateVersionViews = certificateVersionViews
        this.signedBy = signedBy
        this.signs = signs
    }
}
